test/core/GenArt721CoreV2_IYK_Integration.tests.ts

```diff
@@ -0,0 +1,295 @@
+import { expect } from "chai";
+import { Wallet } from "ethers";
+import { arrayify } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+import {
+  assignDefaultConstants,
+  deployAndGet,
+  getAccounts,
+} from "../util/common";
+
+/**
+ * These tests are intended to check the IYK integration's functionality.
+ * These tests are not complete, and assume all GenArt721CoreV2_PBAB functionality
+ * remains perfectly function, as GenArt721CoreV2_IYK subclasses it.
+ */
+describe("GenArt721CoreV2_IYK_Integration", async function () {
+  beforeEach(async function () {
+    // standard accounts and constants
+    this.accounts = await getAccounts();
+    await assignDefaultConstants.call(this);
+    this.signVerifier = new Wallet(
+      "fb45bc2049e143fb168fd438ae6dd3eefffa7f798a7f4d865b6748831abaa625" as string
+    );
+    // deploy and configure core, randomizer, and minter
+    this.randomizer = await deployAndGet.call(this, "BasicRandomizer", []);
+    // V2_PRTNR need additional arg for starting project ID
+    this.genArt721Core = await deployAndGet.call(this, "GenArt721CoreV2_IYK", [
+      this.name,
+      this.symbol,
+      this.randomizer.address,
+      0,
+    ]);
+    this.minter = await deployAndGet.call(this, "GenArt721Minter_PBAB", [
+      this.genArt721Core.address,
+    ]);
+
+    // add minter
+    await this.genArt721Core
+      .connect(this.accounts.deployer)
+      .addMintWhitelisted(this.minter.address);
+
+    // add project
+    await this.genArt721Core
+      .connect(this.accounts.deployer)
+      .addProject("name", this.accounts.artist.address, this.projectZero);
+
+    // Assign iyk verifier
+    await this.genArt721Core
+      .connect(this.accounts.deployer)
+      .setSignVerifier(this.signVerifier.address);
+  });
+
+  describe("getClaimSigningHash", () => {
+    describe("should return the correct hash", () => {
+      it("when called", async function () {
+        const blockExpiry = ethers.BigNumber.from(123);
+        const tokenId = ethers.BigNumber.from(1);
+
+        const claimSigningHash = await this.genArt721Core.getClaimSigningHash(
+          blockExpiry,
+          this.accounts.user.address,
+          tokenId
+        );
+        expect(claimSigningHash).to.equal(
+          ethers.utils.solidityKeccak256(
+            ["uint256", "address", "uint256", "address", "uint256"],
+            [
+              blockExpiry,
+              this.accounts.user.address,
+              tokenId,
+              this.genArt721Core.address,
+              await this.genArt721Core.getClaimNonce(
+                this.accounts.user.address
+              ),
+            ]
+          )
+        );
+      });
+    });
+  });
+
+  describe("claimNFT", () => {
+    describe("should transfer a tokens ownership", () => {
+      it("when the signature is valid", async function () {
+        await this.genArt721Core
```

Comment:
I think these can move into `beforeEach` since the only test that doesn't use them is a view function test for `getClaimSigningHash` above